### DISCUSSION TITLE
Unnecessary file versions

### DISCUSSION THREAD

**Discussion by chrysenteron** [opengisch/QField#3977] [Q&A]
Hi, 
Is there a way to exclude certain files from file versioning or protect them from getting updated respectively. The bulk of my data storage ist taken up by layers that are not editable and therefore should not be updated every single time the project is synchronized, deltas are applied etc. The fact that new (and identical) file versions are produced every time causes my project to quickly fill up the 1 GB storage again and again.

**Comment by suricactus** [comment]
Unfortunately not yet, but it is in our plans for the next months.

**Reply by chrysenteron** [reply]
Great, looking forward to that, thanks for your response :)

**Reply by chrysenteron** [reply]
Is there a list of features, that you are already planning to add? Might be helpful so people can consult it before suggesting features.

**Reply by suricactus** [reply]
There is no such public list yet. You are not alone asking for this, we are evaluating our possibilities. For now, you can follow the QFieldCloud repository and see the pull requests getting merged.

**Reply by chrysenteron** [reply]
Hi, 
Our work was on hold for a bit now, so I didn't follow your work closely the past months, but I expect we'll have to pick it up again quite soon. So I was wondering, if you have made any progress with this (File versioning of files that haven't been changed) in the meantime.

**Reply by suricactus** [reply]
We are making a good progress with the feature to manually set the number of file versions to keep for each project. It will be released in the upcoming versions.

> File versioning of files that haven't been changed

Can you give more concrete examples? Gpkgs that have been opened for editing are a known issue and there is hardly any workaround. If you experience with other file formats, please let us know.

**Reply by chrysenteron** [reply]
It's been a while, but at the moment, I have some problems with the file versioning again. I get new versions of certain gpkg files, when synchronizing, even though geometries have been locked for the associated layers. Also, I get multiple versions of several tif files, which doesn't make sense, since these can't be edited either. The 1 GB Quota of my organization would be enough, if only the actually editable FIles got updated, but it gets eaten up by additional versions of non-editable files quite quickly.
I hope, there's some kind of solution to this apart from adding more storage. Thanks in advance

**Comment by ZsanettMed** [comment]
Hello @chrysenteron!
 
This idea has been posted on our newly activated [QField Idea Platform](https://ideas.qfield.org/qfieldcloud-feature-requests/p/selective-file-versioning-for-storage-optimization). This platform allows you and other users to upvote ideas, helping us prioritize which features to develop next. The more upvotes an idea gets, the closer it moves towards implementation. :) You can also track the progress of your idea directly on the platform. Thank you for contributing to the continuous improvement of QFieldCloud!

Best Regards!
Zsanett
